controllers/dogu_manager.go

```diff
@@ -91,17 +109,51 @@ func (m DoguManager) Install(ctx context.Context, doguResource *k8sv1.Dogu) erro
 		return fmt.Errorf("failed to create dogu resources: %w", err)
 	}
 
-	logger.Info("Add dogu finalizer...")
-	controllerutil.AddFinalizer(doguResource, finalizerName)
-	err = m.Client.Update(ctx, doguResource)
+	doguResource.Status = k8sv1.DoguStatus{Status: k8sv1.DoguStatusInstalled}
+	err = m.Client.Status().Update(ctx, doguResource)
 	if err != nil {
-		logger.Info(fmt.Sprintf("Dogu %s/%s has been : %s", doguResource.Namespace, doguResource.Name, controllerutil.OperationResultUpdated))
-		return fmt.Errorf("failed to update dogu: %w", err)
+		return fmt.Errorf("failed to update dogu status: %w", err)
+	}
+
+	if customConfigMap != nil {
+		err = m.Client.Delete(ctx, customConfigMap)
+		if err != nil {
+			return fmt.Errorf("failed to delete custom dogu descriptor: %w", err)
+		}
 	}
 
 	return nil
 }
 
+func (m DoguManager) getDoguDescriptor(ctx context.Context, doguResource *k8sv1.Dogu) (*core.Dogu, *corev1.ConfigMap, error) {
+	logger := log.FromContext(ctx)
+	var dogu = &core.Dogu{}
+	configMap := &corev1.ConfigMap{}
+	err := m.Client.Get(ctx, doguResource.GetDescriptorObjectKey(), configMap)
+
+	if err != nil {
```

Comment:
Splitting into integration methods here because two different things happen: Pick either the regular config map or the custom dogu.json